ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlBareOutputNode.java

```diff
@@ -15,25 +15,32 @@
 
 package io.confluent.ksql.planner.plan;
 
+import static java.util.Objects.requireNonNull;
+
 import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
 import io.confluent.ksql.execution.timestamp.TimestampColumn;
 import io.confluent.ksql.name.SourceName;
 import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.serde.WindowInfo;
 import io.confluent.ksql.structured.SchemaKStream;
 import java.util.Optional;
 import java.util.OptionalInt;
 
 @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
 public class KsqlBareOutputNode extends OutputNode {
 
+  private final Optional<WindowInfo> windowInfo;
+
   public KsqlBareOutputNode(
       final PlanNodeId id,
       final PlanNode source,
       final LogicalSchema schema,
       final OptionalInt limit,
-      final Optional<TimestampColumn> timestampColumn
+      final Optional<TimestampColumn> timestampColumn,
+      final Optional<WindowInfo> windowInfo
```

Comment:
To check my understanding: the reason this PR adds window info into the output node is so the window info can be printed as part of the key?

Reply:
It's needed so that the key schema, returned at the start of a push query, can include the window bounds columns for windowed results.